include/networkit/reachability/ReachableNodes.hpp

```diff
@@ -0,0 +1,80 @@
+// networkit-format
+
+#ifndef NETWORKIT_REACHABILITY_REACHABLE_NODES_HPP_
+#define NETWORKIT_REACHABILITY_REACHABLE_NODES_HPP_
+
+#include <vector>
+
+#include <networkit/base/Algorithm.hpp>
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+
+/**
+ * @ingroup reachability
+ */
+class ReachableNodes final : public Algorithm {
+public:
+    /**
+     * Determines or estimates the number of reachable nodes from each node in the graph.
+     *
+     * @param G The graph.
+     * @param exact Whether or not to compute the number of reachable nodes exactly. Only used for
+     * directed graphs, on undirected graphs the number of reachable nodes from every node can be
+     * computed in linear time.
+     */
+    ReachableNodes(const Graph &G, bool exact = true);
```

Comment:
Since the `run()` method can be called repeatedly, how about adding a setter for `exact` (or exposing the variable)? In this way a user doesn't have to allocate 2*n memory a second time for having the same run with exact/non-exact.

Reply:
Yes I think that's reasonable.